database/src/main/java/com/firebase/ui/database/FirebaseIndexArray.java

```diff
@@ -155,10 +155,11 @@ public void onDataChange(DataSnapshot snapshot) {
                     notifyChangedListeners(OnChangedListener.EventType.CHANGED, index);
                 }
             } else {
-                Log.w(TAG, "Key not found at ref: " + snapshot.getRef());
                 if (isMatch(index, key)) {
                     mDataSnapshots.remove(index);
                     notifyChangedListeners(OnChangedListener.EventType.REMOVED, index);
+                } else {
+                    Log.w(TAG, "Key not found at ref: " + snapshot.getRef());
```

Comment:
Only log the warning if the data wasn't in our list. Example: if people delete the data ref before the index ref, the warning would have been shown which is kinda pointless.